service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpFeedbackReceivedHandler.java

```diff
@@ -93,6 +116,26 @@ public AmqpFeedbackReceivedHandler(String hostName, String userName, String sasT
         add(new FlowController());
     }
 
+    AmqpFeedbackReceivedHandler(
+            String hostName,
+            TokenCredential authenticationTokenProvider,
+            TokenCredentialType authorizationType,
+            IotHubServiceClientProtocol iotHubServiceClientProtocol,
+            AmqpFeedbackReceivedEvent amqpFeedbackReceivedEvent,
+            ProxyOptions proxyOptions,
+            SSLContext sslContext)
+    {
+        super(hostName, authenticationTokenProvider, authorizationType, iotHubServiceClientProtocol, proxyOptions, sslContext);
+
+        this.amqpFeedbackReceivedEvent = amqpFeedbackReceivedEvent;
+
+        // Add a child handler that performs some default handshaking
```

Comment:
What is this? More for my understanding. If it makes sense, you can add more comments in code too.

Reply:
It's a cool little handler that proton-j gives us. From their documentation:

```java
/**
 * A handler that mirrors the actions of the remote end of a connection.  This
 * handler responds in kind when the remote end of the connection is opened and
 * closed.  Likewise if the remote end of the connection opens or closes
 * sessions and links, this handler responds by opening or closing the local end
 * of the session or link.
 */
```

Reply:
I've also added this as a comment next to where we create these handshakers for future reference